specs/AppNotifications/AppNotificationContentSpec/AppNotificationBuilder-spec.md

```diff
@@ -0,0 +1,1201 @@
+data.Value(1.0); // Binds to {progressValue} in xml payload
+data.ValueStringOverride(L"2/2 songs"); // Binds to {progressValueString} in xml payload
+data.Status(L"Downloaded!"); // Binds to {progressStatus} in xml payload
+
+auto result = co_await winrt::Microsoft::Windows::AppNotifications::AppNotificationManager::Default().UpdateAsync(data, L"Tag", L"Group");
+```
+
+Example result: ![Progress Bar Example 2](ProgressBarExample2.png)
+
+# Retrieving Arguments
+
+AppNotificationContent and Button return arguments to the activated application when the user clicks
+on the component. Developers can add multiple key/value pairs and Windows App SDK will serialize
+them into a string.
+
+Below is an example with AppNotificationContent and Button:
+
+```c#
+AppNotificationContent(
+    .AddArgument(L"action", L"appNotificationBodyTouch")
+    .AddArgument(L"user", 938163)
+    .AddButton(Button(L"Send")
+        .AddArgument(L"action", L"buttonTouch")
+        .AddArgument(L"user", 938163)));
+```
+
+When the user clicks on the AppNotification body, the resulting argument string is:
+
+-   "action=appNotificationBodyTouch;user=938163"
+
+When the user clicks on the Button, the result argument string is:
+
+-   "action=buttonTouch;callId=938163"
+
+Notice that ';' is used as the delimiter between each argument.
+
+To retrieve the arguments from the serialized string, developers will use
+AppNotificationReceivedEventArgs::Arguments to get the values from the argument string as a map.
+
+Below is an example usage:
+
+```c#
+using namespace winrt::Microsoft::Windows::AppLifecycle;
+using namespace winrt::Microsoft::Windows::AppNotifications;
+
+// Retrieve the activation args using Windows App SDK AppLifecycle functions
+AppActivationArguments args { AppLifecycle::AppInstance::GetCurrent().GetActivatedEventArgs() };
+ExtendedActivationKind kind { args.Kind() };
+
+// Check the activation kind
+if (kind == ExtendedActivationKind::AppNotification)
+{
+    AppNotificationActivatedEventArgs appActivatedArgs { args.Data().as<AppNotificationActivatedEventArgs>() };
+
+    // Retrieve the serialized arguments from AppNotificationActivatedEventArgs
+    winrt::Windows::Foundation::Collections::IMap<winrt::hstring, winrt::hstring> argumentMap { appActivatedArgs.Argument() };
+}
+```
+
+# Full API Details
+
+```c#
+namespace Microsoft.Windows.AppNotifications.Builder
+{
+    runtimeclass AppNotificationTextPropertiesGreen
+    {
+        // Contains the set of <text> attributes
+        AppNotificationTextProperties();
+
+        AppNotificationTextProperties SetLanguage(String value);
+        AppNotificationTextProperties UsesCallScenarioAlign();
+        AppNotificationTextProperties SetMaxLines(Int32 value);
+
+        String GetXml();
+    }
+
+    enum ButtonColor
+    {
+        Red, // The button will be green
+        Green, // The button will be red
+    };
+
+    runtimeclass AppNotificationButton
```

Comment:
I would prefer this actually had an event on it that would get called when the user clicks it, instead of having to go all the way through activation events (of course only used by actively running apps, but it would likely already be running if using this).